src/library/dlgtrackinfo.cpp

```diff
@@ -596,32 +596,23 @@ void DlgTrackInfo::slotKeyTextChanged() {
 
     // If the new key string is invalid and not empty them reject the new key.
     if (globalKey == mixxx::track::io::key::INVALID && !newKeyText.isEmpty()) {
-        txtKey->setText(KeyUtils::getGlobalKeyText(m_keysClone));
+        txtKey->setText(KeyUtils::getGlobalKeyText(m_keys));
         return;
     }
 
     // If the new key is the same as the old key, reject the change.
-    if (globalKey == m_keysClone.getGlobalKey()) {
+    if (globalKey == m_keys.getGlobalKey()) {
         return;
     }
 
     // Otherwise, accept.
-    m_keysClone = newKeys;
+    m_keys = newKeys;
 }
 
 void DlgTrackInfo::reloadTrackMetadata() {
     if (m_pLoadedTrack) {
-        // Allocate a temporary track object for reading the metadata.
-        // We cannot reuse m_pLoadedTrack, because it might already been
-        // modified and we want to read fresh metadata directly from the
-        // file. Otherwise the changes in m_pLoadedTrack would be lost.
-        TrackPointer pTrack(Track::newTemporary(
-                m_pLoadedTrack->getFileInfo(),
-                m_pLoadedTrack->getSecurityToken()));
-        SoundSourceProxy(pTrack).loadTrackMetadata();
-        if (!pTrack.isNull()) {
-            populateFields(*pTrack);
-        }
+        SoundSourceProxy(m_pLoadedTrack).loadTrackMetadata(true);
```

Comment:
This PR just eliminates the inconsistencies and establishes a consistent 
behaviour for both reloading and fetching metadata.

The whole dialog is broken by design with all the slot/signal connections of 
the loaded track and its contained objects like beat grid and cue point. 
Especially fetching metadata from MusicBrainz is currently tied to the 
loaded track object and relies on signals emitted by this object to receive 
the results.

On 30.06.2016 11:32, Daniel Schürmann wrote:
>
> In src/library/dlgtrackinfo.cpp 
> <https://github.com/mixxxdj/mixxx/pull/973#discussion_r69101771>:
>
> >  }
> >
> >  void DlgTrackInfo::reloadTrackMetadata() {
> >      if (m_pLoadedTrack) {
> > -        // Allocate a temporary track object for reading the metadata.
> > -        // We cannot reuse m_pLoadedTrack, because it might already been
> > -        // modified and we want to read fresh metadata directly from the
> > -        // file. Otherwise the changes in m_pLoadedTrack would be lost.
> > -        TrackPointer pTrack(Track::newTemporary(
> > -                m_pLoadedTrack->getFileInfo(),
> > -                m_pLoadedTrack->getSecurityToken()));
> > -        SoundSourceProxy(pTrack).loadTrackMetadata();
> > -        if (!pTrack.isNull()) {
> > -            populateFields(*pTrack);
> > -        }
> > +        SoundSourceProxy(m_pLoadedTrack).loadTrackMetadata(true);
>
> IMHO the comment and the underlying issue still applies.
>
> m_pLoadedTrack is a global track object, that is used everywhere else.
> If we load the file metadata directly to the track, the changes are 
> applied mediately without a chance for cancel.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/973/files/13403bd2e1641927c46bd3012d177ad7d367b876#r69101771>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/ABZ-apDr19I-BzTIA1I8_AHJut0H5FcIks5qQ41GgaJpZM4JBhOn>.
>